includes/Discovery.php

```diff
@@ -0,0 +1,308 @@
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories;
+
+use Google\Web_Stories\REST_API\Stories_Controller;
+use WP_Post;
+
+/**
+ * Discovery class.
+ */
+class Discovery {
+	/**
+	 * Initialize discovery functionality.
+	 *
+	 * @return void
+	 */
+	public function init() {
+		add_action(
+			'web_stories_story_head',
+			static function () {
+				// Theme support for title-tag is implied for stories. See _wp_render_title_tag().
+				echo '<title>' . esc_html( wp_get_document_title() ) . '</title>' . "\n";
+			},
+			1
+		);
+
+		add_action( 'web_stories_story_head', [ $this, 'print_schemaorg_metadata' ] );
+		add_action( 'web_stories_story_head', [ $this, 'print_open_graph_metadata' ] );
+		add_action( 'web_stories_story_head', [ $this, 'print_twitter_metadata' ] );
+
+		// @todo Check if there's something to skip in the new version.
+		add_action( 'web_stories_story_head', 'rest_output_link_wp_head', 10, 0 );
+		add_action( 'web_stories_story_head', 'wp_resource_hints', 2 );
+		add_action( 'web_stories_story_head', 'feed_links', 2 );
+		add_action( 'web_stories_story_head', 'feed_links_extra', 3 );
+		add_action( 'web_stories_story_head', 'rsd_link' );
+		add_action( 'web_stories_story_head', 'wlwmanifest_link' );
+		add_action( 'web_stories_story_head', 'adjacent_posts_rel_link_wp_head', 10, 0 );
+		add_action( 'web_stories_story_head', 'noindex', 1 );
+		add_action( 'web_stories_story_head', 'wp_generator' );
+		add_action( 'web_stories_story_head', 'rel_canonical' );
+		add_action( 'web_stories_story_head', 'wp_shortlink_wp_head', 10, 0 );
+		add_action( 'web_stories_story_head', 'wp_site_icon', 99 );
+		add_action( 'web_stories_story_head', 'wp_oembed_add_discovery_links' );
+	}
+
+	/**
+	 * Prints the schema.org metadata on the single story template.
+	 *
+	 * @return void
+	 */
+	public function print_schemaorg_metadata() {
+		$metadata = $this->get_schemaorg_metadata();
+
+		?>
+		<script type="application/ld+json"><?php echo wp_json_encode( $metadata, JSON_UNESCAPED_UNICODE ); ?></script>
+		<?php
+	}
+
+	/**
+	 * Get schema.org metadata for the current query.
+	 *
+	 * @return array $metadata All schema.org metadata for the post.
+	 */
+	protected function get_schemaorg_metadata() {
```

Comment:
Seems like this is copied from - https://github.com/ampproject/amp-wp/blob/722f8c9612b2e831217ed9e72ba50caa421f3553/includes/amp-helper-functions.php#L1526-L1594

Reply:
Originally, yes. But in this PR I merely moved the function around.

The metadata format will probably change slightly. For example, `BlogPosting` is not entirely accurate for a story.

Reply:
Just wonder if it worth having a comment to say that it was taken from there.